ci/Vagrantfile

```diff
@@ -0,0 +1,41 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+require 'etc'
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "generic/ubuntu1804"
+if Vagrant.has_plugin?('vagrant-env')
+    config.env.enable
+  end
+
+  machine_name = ENV['MACHINE_NAME'] || ("vmck-vagrant-" + `hostname`.strip)
+  custom_sh = ENV['PROVISION_SH']
+
+  config.vm.define machine_name
+  # disable implicit mount for /vagrant
+  config.vm.synced_folder '.', '/vagrant', disabled: true
+
+  # blank provisioner that can be overridden by providers
+  config.vm.provision 'provider', type: 'shell', inline: '/bin/true', privileged: false
+
+  config.vm.provider :vmck do |vmck, override|
```

Comment:
You can massively simplify this vagrantfile: move overrides to above and only keep `vmck`-specific settings here, because we only need the vmck backend for the vmck test on vmck.